src/components/ProfileTabbedView.tsx

```diff
@@ -0,0 +1,68 @@
+import { Tab, TabList, TabPanel, Tabs } from "react-tabs";
+import {useState} from "react";
+import {api} from "~/utils/api";
+import {useRouter} from "next/router";
+import {ProfileCard} from "./ProfileCard";
+import { Spinner } from "./Spinner";
+import {BlogPostViewer} from "./BlogPostViewer";
+import {Layout} from "./Layout";
+
+export const ProfileTabbedView: React.FC = () => {
+    const [_, setForceRerender] = useState(false);
+    const forceRerender = () => setForceRerender(!_);
+    const [tabState, setTabState] = useState(0);
+
+    const router = useRouter();
+    const { id } = router.query;
+    if(!router.isReady) {
+        return (<Spinner size={2} />);
+    }
+    const { data: userData } = api.user.selectedUser.useQuery({
```

Comment:
Getting an error when directly navigating to the url (go to _http://localhost:3000/profile/clfot8hi60000f8kkzngxn8qq_ for example) :

```
Unhandled Runtime Error
Error: Rendered more hooks than during the previous render.

Source
src\components\ProfileTabbedView.tsx (20:31) @ api

  18 |     return (<Spinner size={2} />);
  19 | }
> 20 | const { data: userData } = api.user.selectedUser.useQuery({
     |                           ^
  21 |     text: {id}.id as string,
  22 | });
  23 | if (!userData) {
```

Reply:
This is probably because you are putting the useQuery hook after you return (if router is not ready), should be an easy fix!

Reply:
after the refactoring, I haven't seen this error, so I believe this is fixed